### ISSUE TITLE
Tracking issue for ASL v1 bugs and to be implemented

### ISSUE THREAD

**Issue by HadrienRenaud** [HadrienRenaud/herdtools7#21]
General features:
- [ ] Nested bitfields
- [ ] Global ignored constants
- [ ] Guards on pattern-matching (at least in v0)
- [ ] `DIVRM`
- [ ] More integer operations in StaticInterpreter (`MOD`, `<<`, `DIV` ...)
- [ ] Conditional expressions in constants
- [ ] Arrays indexed by enumerations
- [ ] Reals
- [ ] Not yet implemented in ASLv1 Parser
- [ ] Not yet implemented in ASLv0 Parser

Type-checking & safety:
- [ ] Runtime checks
- [ ] Overlapping in bitfield slices
- [ ] Division by zero
- [ ] Some constrained integers refinements are needed, but maybe blocked on soundness of type-system
- [ ] Check for initialization value of bitvector lengths
- [ ] Check for compile-time evaluation of constants
- [ ] Getters without args
- [ ] Parameter unification
- [ ] Check that parameters have corresponding formal
- [ ] Parameter type-check
- [ ] Check that patterns are constant expressions
- [ ] Reserved keywords (low priority)

**Comment by HadrienRenaud** [comment]
After investigation and further discussion, "Merge ASLScalar and constants in aslseq" looks complicated and without any clear improvements in sight. Dropping.
